doc/design/doi-generation-1.md

```diff
@@ -0,0 +1,26 @@
+# DOI generation process
```

Comment:
```suggestion
# Publish process

This process is predicated on a few tasks that are expected to receive valid input and assuming that only certain errors could happen that are out of our control.

- publish enabled only if dandiset is valid (so all validation checks have passed)
- on publish trigger:
   - create the versioned release in the db (this prevents any asset/blob belonging to the release from being deleted)
   - insert all the relevant metadata (version, doi, url, manifest location/checksum) into the metadata record of the new version 
  - send this new metadata record to mint the doi (there should be no reason why this should fail unless the external resource fails. if it fails notify admins since this is likely a new bug)
   - write out to s3 bucket (even on any datacite failure)
   - switch over ownership of the release to dandi (unless releases are not modifiable by the dandiset creator)
   
## DOI generation process
```

Reply:
I thought this PR would be scoped to DOIs so Dorota and I could get a cohesive design. If you want to cover the entire publish process, I can add:
* Metadata validation (so the publish button isn't available if metadata is invalid)
* Web GUI UX
* Creating the published version
* Writing the manifests

Reply:
yes let's scope the whole thing. because the datacite conversion and doi generation sits within the context of that.